update.go

```diff
@@ -394,23 +394,39 @@ func uploadTemp(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type divertRequest struct {
+	Path      string   `json:"Path"`
+	Diversion string   `json:"Diversion"`
+	Flags     []string `json:"Flags"`
+}
+
 func divert(w http.ResponseWriter, r *http.Request) {
 	if r.Method != http.MethodPost {
 		http.Error(w, "expected a POST request", http.StatusBadRequest)
 		return
 	}
 
-	diversion := r.FormValue("diversion")
-	if diversion == "" {
+	var req divertRequest
+	if r.Header.Get("Content-Type") == "application/json" {
+		err := json.NewDecoder(r.Body).Decode(&req)
```

Comment:
This is a common mistake: json.NewDecoder “reads and decodes JSON values from an input stream.”, but we don’t have an input stream — we only have a single value. It’s better to just call json.Unmarshal() on the result of io.ReadAll(resp.Body) in this case.

Reply:
Fixed!